NetworkingSampleApp/NetworkingSampleApp/Scenes/Authorization/AuthorizationViewModel.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  AuthorizationViewModel.swift
+//  NetworkingSampleApp
+//
+//  Created by Dominika Gajdová on 06.12.2022.
+//
+
+import Foundation
+import Networking
+import OSLog
+
+final class AuthorizationViewModel: ObservableObject {
+    private lazy var authManager = SampleAuthorizationManager()
+    private lazy var apiManager: APIManager = {
+        let authorizationInterceptor = AuthorizationTokenInterceptor(authorizationManager: authManager)
+        
+        var responseProcessors: [ResponseProcessing] = [
+            LoggingInterceptor.shared,
+            authorizationInterceptor,
+            StatusCodeProcessor.shared
+        ]
+        var errorProcessors: [ErrorProcessing] = [LoggingInterceptor.shared]
+        
+#if DEBUG
+        responseProcessors.append(EndpointRequestStorageProcessor.shared)
+        errorProcessors.append(EndpointRequestStorageProcessor.shared)
+#endif
+        
+        return APIManager(
+            responseProvider: MockResponseProvider(with: Bundle.main, sessionId: "2023-01-31T15:08:08Z"),
+            requestAdapters: [
+                LoggingInterceptor.shared,
+                authorizationInterceptor
+            ],
+            responseProcessors: responseProcessors,
+            errorProcessors: errorProcessors
+        )
+    }()
+}
+
+extension AuthorizationViewModel {
+    func loadUserList() async throws {
+        try await apiManager.request(
+            SampleUserRouter.users(page: 2)
+        )
+    }
+    
+    func login(email: String?, password: String?) async throws {
+        let request = SampleUserAuthRequest(email: email, password: password)
+        let response: SampleUserAuthResponse = try await apiManager.request(
+            SampleAuthRouter.loginUser(request)
+        )
+        
+        let data = response.authData
+        // Save login token data to auth storage.
+        try await authManager.storage.save(data: data)
```

Comment:
Shouldn't this be hidden inside our implementation? Vs view model is handling this

Reply:
I agree, just not sure what would be a better approach. Here is a discussion about this https://github.com/strvcom/ios-networking/pull/24#discussion_r1093325815

Reply:
this is related to one open topic, that first time you need to save manually than it's automatically by auth layer